pkgs/os-specific/linux/cpupower/gui.nix

```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchFromGitHub
+, desktop-file-utils
+, gettext
+, glib
+, gobject-introspection
+, gtk3
+, hicolor-icon-theme
+, meson
+, ninja
+, pkg-config
+, python3
+, systemd
+, wrapGAppsHook
+}:
+
+let
+  # use pythonEnv, so that patchShebang fixup will choose this interpreter
+  pythonEnv = python3.withPackages(ps: with ps; [
+    pygobject3
+    dbus-python
+  ]);
+in
+
+stdenv.mkDerivation rec {
+  pname = "cpupower-gui";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "vagnum08";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1rrzlcmspill7qivv3bg8v810ppj4wxa9jcf2ddrjqcbq4lfxayk";
+  };
+
+  nativeBuildInputs = [
+    hicolor-icon-theme # needed for postinstall script
+    desktop-file-utils # needed for update-desktop-database
+    meson
+    ninja
+    glib # needed for glib-compile-scemas
+    pkg-config
+    wrapGAppsHook
+    gettext
+    pythonEnv
+    gtk3
+    gobject-introspection # need for gtk namespace to be available
+  ];
+
+  mesonFlags = [
+    "-Dsystemddir=${placeholder "out"}/lib/systemd"
+  ];
+
+  preConfigure = ''
+    patchShebangs ./build-aux/meson/postinstall.py
+  '';
+
+  # Polkit rules aren't installed and we need them to authorize users via polkit
+  # we refer to this file in the corresponding nixos module
+  postInstall = ''
+    mkdir ${placeholder "out"}/share/polkit-1/rules/
+    cp ../data/polkit/cpupower-gui.rules ${placeholder "out"}/share/polkit-1/rules
```

Comment:
can you install these to `etc/polkit-1/rules.d/cpupower-gui.rules`? and also purge `.pkla` from the output? It appears to read them from multiple locations https://www.freedesktop.org/software/polkit/docs/latest/polkit.8.html, and I think the supported one on NixOS (IIRC) is at `etc/`. For consistency we should install it there as well

Reply:
we could also open an issue upstream that `.pkla` is deprecated and why don't they install the `.rules`? I can't even think of distros that use a `sudo` group either.

Reply:
See https://github.com/vagnum08/cpupower-gui/issues/28 . I asked before about  `.rules` not being installed.
I also install to `/etc/polkit-1` but through the nixos module. Was also wondering about an option to control if these rules are installed/enabled or not.

Reply:
Added deletion of pkla file in `${out}/var...`

Reply:
About installing to `\etc` I was referring to https://github.com/NixOS/nixpkgs/blob/2ff45e726e1871aaf84039efb0d6acb89eaa7be1/nixos/modules/services/desktops/cpupower-gui.nix#L23 in the nixos module

Reply:
> I also install to /etc/polkit-1 but through the nixos module. Was also wondering about an option to control if these rules are installed/enabled or not.

What i meant was this is how polkit loads these rules:
![Screenshot from 2020-10-11 22-42-49](https://user-images.githubusercontent.com/28888242/95699800-76aea100-0c13-11eb-87b9-20b54d0db40a.png)


polkit is configured with `"--datadir=${system}/share"`, ${system} being `/run/current-system/sw`.
So I think it should be able to load these rules by virtue of it being in systemPackages, no?
And it also seem you have the path incorrect as `share/polkit-1/rules` when it should be `share/polkit-1/rules.d`.

Reply:
You are right!. Fixing the typo also avoids the need for workarounds in the nixos module.
The rules file still needs to be copied manually though, at least until fixed upstream.

I also didn't have to specify `--datadir` as the default seems to do the right thing.
Pull request now includes this correction..

Reply:
Actually, I spoke too soon. After removing the manual copy to `/etc` I started getting a dbus error. Turns out the `pkla` is still necessary.

Reply:
Puzzled... Got the same error again but now the pkla is in place. It was working a minute ago...
I'm going to give this a rest and try again tomorrow. The error is a bit cryptic as well:
```
Traceback (most recent call last):
  File "/nix/store/bz7r7v2plyahszyydb6kilnjp9n60n7r-cpupower-gui-0.8.0/share/cpupower-gui/cpupower_gui/window.py", line 205, in on_apply_clicked
    if HELPER.isauthorized():
  File "/nix/store/8zik6jm24nv53khsfkj50fy0a2ffs1mc-python3.7-dbus-python-1.2.12/lib/python3.7/site-packages/dbus/proxies.py", line 147, in __call__
    **keywords)
  File "/nix/store/8zik6jm24nv53khsfkj50fy0a2ffs1mc-python3.7-dbus-python-1.2.12/lib/python3.7/site-packages/dbus/connection.py", line 653, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name :1.3551 was not provided by any .service files
```

Reply:
This error can occur when the polkit daemon has been restarted, and the helper service is using the old connection.

As an example, for this specific error the `org.freedesktop.PolicyKit1.Authority` had the dbus name `:1.3551` (where `3551` I think is the PID but we don't have to worry about it). When `polkit` restarted, it got a different name  but the cpupower-helper daemon was using the old one.

I will have a fix in the next release (0.9.0).

Reply:
> Actually, I spoke too soon. After removing the manual copy to `/etc` I started getting a dbus error. Turns out the `pkla` is still necessary.

The `.pkla` is not necessary. I have a guess what might be going one.

1. We know that the rules file is correct and works as expected when install in `/etc/`.
2. The reason that it might not work when installed in `/usr/share/polkit-1/rules.d` is the permission of `rules.d` directory.

 I did a test in an Arch Linux install and the permissions of `/usr/share/polkit-1` were:
```
 sudo ls -la /usr/share/polkit-1
total 0
drwxr-xr-x 1 root root      28 Dec 13  2016 .
drwxr-xr-x 1 root root    9314 Oct 12 13:58 ..
drwxr-xr-x 1 root root    3978 Oct 12 14:00 actions
drwxr-x--- 1 root root     462 Oct 12 14:00 rules.d
```
I found a bug report from 2017, https://bugs.archlinux.org/task/54604.
> systemd (as of now, from [testing]) version 233-5 changes the group of /usr/share/polkit-1/rules.d/ to root:root, while the
directory should really be group polkitd:
> this change makes polkit to not be able to read /usr/share/polkit-1/rules.d/systemd-networkd.rules

Apparently, the change needed manual user intervention to reset the permissions.
I noticed that the packages (on Arch) that use polkit (including the systemd package) run `install -d -o root -g 102 -m 750 "$pkgdir/usr/share/polkit-1/rules.d"` during packaging. The flag `-g 102` sets the group to polkitd.

Knowing the above, I run the following commands, and now the rules file works as expected.

```bash
$ sudo chown :polkitd /usr/share/polkit-1/rules.d/
$ sudo ls -la /usr/share/polkit-1
total 0
drwxr-xr-x 1 root root      28 Dec 13  2016 .
drwxr-xr-x 1 root root    9314 Oct 12 13:58 ..
drwxr-xr-x 1 root root    3978 Oct 12 14:00 actions
drwxr-x--- 1 root polkitd  462 Oct 12 14:00 rules.d
```

PS. I will have meson set the correct permissions in the next release.

Reply:
Ah thanks! This explains why it was intermittently working.
Today after a system reboot all seems to be working fine with the version in the pull request.
@vagnum08 thanks for the update. Looking forward to 0.9.0.